### PR TITLE
feat: read token with selected rules

### PR DESCRIPTION
**Pull request by dimabru**

### CODE REVIEW

File: pkg/rulesConfig/rulesConfig.go [hunk 1]
```diff
@@ -162,6 +171,40 @@ func (rc *RulesConfig) GetAllRuleNames() []string {
 	return ruleNames
 }
 
+func (rc *RulesConfig) GetSelectedRuleIds(ignoreToken bool) (map[int]bool, error) {
+	if ignoreToken {
```
Comment (OriYosef) [inline pkg/rulesConfig/rulesConfig.go:175]: why not move the if to the func call?

File: pkg/rulesConfig/rulesConfig.go [hunk 2]
```diff
@@ -162,6 +171,40 @@ func (rc *RulesConfig) GetAllRuleNames() []string {
...
+	if err != nil {
+		return nil, err
+	}
+	if token == nil {
+		return nil, nil
+	}
+
+	rawDecodedToken, err := base64.StdEncoding.DecodeString(fmt.Sprintf("%v", token))
+	if err != nil {
+		return nil, fmt.Errorf("error decoding token. run `allero config clear token` to clear the existing token")
```
Comment (OriYosef) [inline pkg/rulesConfig/rulesConfig.go:189]: maybe add:
"and generate a new one. {link}"

File: pkg/rulesConfig/rulesConfig.go [hunk 3]
```diff
@@ -162,6 +171,40 @@ func (rc *RulesConfig) GetAllRuleNames() []string {
...
+
+	decodedToken := &DecodedToken{}
+	err = json.Unmarshal(rawDecodedToken, decodedToken)
+	if err != nil {
+		return nil, err
+	}
+
+	selectedRuleIds := make(map[int]bool)
+	for i, rule := range decodedToken.Rules {
+		if rule {
```
Comment (OriYosef) [inline pkg/rulesConfig/rulesConfig.go:200]: `selectedRuleIds[i+1] = rule`

is enough. No?

**Review by OriYosef — APPROVED** [review]

File: pkg/rulesConfig/rulesConfig.go (hunk 1 above)
Comment (dimabru) [reply pkg/rulesConfig/rulesConfig.go:175]: moved outside of the function call

**Pull request merged**